core/common/src/main/java/alluxio/collections/FieldIndex.java

```diff
@@ -61,4 +70,20 @@
    * @return the object or null if there is no such object
    */
   T getFirst(Object value);
+
+  /**
+   * Returns an iterator over the elements in this index. The elements are returned in no particular
+   * order.
+   *
+   * Note that the behavior of the iterator is unspecified if the underlying collection is modified
+   * while a thread is going through the iterator.
+   *
+   * @return an iterator over the elements in this {@link FieldIndex}
+   */
+  Iterator<T> iterator();
+
+  /**
+   * @return the number of objects in this indexed set (O(1) time)
```

Comment:
why is O(1) in the comment? Does this mean all implementations must have an O(1) implementation?